code/datums/components/growth_and_differentiation.dm

```diff
@@ -52,11 +65,19 @@
 	return setup_growth_tracking()
 
 /datum/component/growth_and_differentiation/Destroy(force, silent)
-	. = ..()
 	deltimer(timer_id)
+	return ..()
 
 /datum/component/growth_and_differentiation/UnregisterFromParent()
 	UnregisterSignal(SSticker, COMSIG_TICKER_ROUND_STARTING)
+	for(var/signal in signals_to_kill_on)
+		UnregisterSignal(parent, signal)
+
+/// In case the mob decides that we shouldn't grow anymore (permanently), we don't really wanna waste up the tick doing useless work. Let's stop everything and qdel ourselves.
+/datum/component/growth_and_differentiation/proc/kill_processes()
+	UnregisterFromParent()
+	STOP_PROCESSING(SSdcs, src)
+	qdel(src)
```

Comment:
`UnregisterFromParent` is called from `Destroy` anyways
Likewise `Destroy` should have the code to stop processing